### PR TITLE
Tweaks to component lifecycle guide

### PR DESCRIPTION
**Pull request by tomdale**
- General language/clarity cleanup
- Move from numeric IDs to semantic IDs (this is more friendly to
  refactorings in the future)
- Remove mention of `_super()` - `didInsertElement` should not require
  this on Ember.Component subclasses. Subclasses of other components
  should follow the normal rules of `_super()` invocation

### CODE REVIEW

File: source/components/the-component-lifecycle.md [hunk 1]
```diff
@@ -1,62 +1,118 @@
-Part of what makes a component such a useful tool is that it is closely tied to your app's templates and the DOM itself.
+Part of what makes a component such a useful tool is that it allows you
```
Comment (michaelrkn) [inline source/components/the-component-lifecycle.md]: "a component" -> "components" (to match "They" in the following sentence)

File: source/components/the-component-lifecycle.md [hunk 2]
```diff
@@ -1,62 +1,118 @@
...
+  this.$('input.date').myDatePickerLib();
+}
+```
+
+[`didInsertElement()`][did-insert-element] is also a good place to
+attach event listeners. This is particularly useful for custom events or
+other browser events which do not have a [built-in event
+handler][event-names].
+
+For example, Perhaps you have some custom CSS animations trigger when the component
```
Comment (michaelrkn) [inline source/components/the-component-lifecycle.md]: lowercase Perhaps

**Comment by michaelrkn** [comment]
Thanks for the improvements, @tomdale! I pointed out just a couple small grammar things I noticed.

**Comment by tomdale** [comment]
@michaelrkn Updated to incorporate your feedback

**Pull request merged**